File: swapi-server/src/main/java/de/lkaktus/entities/StarshipDTO.java
package de.lkaktus.entities;

import com.fasterxml.jackson.annotation.JsonProperty;

public record StarshipDTO(
        @JsonProperty("MGLT") String maxSpeed,
        @JsonProperty("cargo_capacity") String cargoCapacity,
        String consumables,
        @JsonProperty("cost_in_credits") String constInCredits,
        String created,
        String crew,
        String edited,
        @JsonProperty("hyperdrive_rating") String hyperdriveRating,
        String length,
        String manufacturer,
        @JsonProperty("max_atmosphering_speed") String maxAtmosphericSpeed,
        String model,
        String name,
        String passengers,
        @JsonProperty("starship_class") String starshipClass,
        String url,
        String [] films,
        SearchItemDTO [] filmItems,
        String [] pilots,
        SearchItemDTO [] pilotItems
) {

}

File: swapi-angular/src/app/api/models/StarshipDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { SearchItemDTO } from './SearchItemDTO';

export type StarshipDTO = {
    MGLT?: string;
    cargo_capacity?: string;
    cost_in_credits?: string;
    hyperdrive_rating?: string;
    maxAtmospheringSpeed?: string;
    starship_class?: string;
    consumables?: string;
    created?: string;
    crew?: string;
    edited?: string;
    length?: string;
    manufacturer?: string;
    model?: string;
    name?: string;
    passengers?: string;
    url?: string;
    films?: Array<string>;
    filmItems?: Array<SearchItemDTO>;
    pilots?: Array<string>;
    pilotItems?: Array<SearchItemDTO>;
};

